Missing includes of some std types

In number of files some includes are missing:

```plain
/root/proj/zkllvm/libs/crypto3/libs/zk/include/nil/crypto3/zk/snark/arithmetization/plonk/variable.hpp:170:26: error: no template named 'map' in namespace 'std'; did you mean 'max'?
                    std::map<typename plonk_variable<AssignmentType>::column_type, std::string> type_map = {
                    ~~~~~^~~
                         max
```

```plain
/root/proj/zkllvm/libs/crypto3/libs/zk/include/nil/crypto3/zk/snark/arithmetization/plonk/constraint.hpp:48:30: error: no template named 'map' in namespace 'std'; did you mean 'max'?
                        std::map<std::tuple<std::size_t, int, typename VariableType::column_type>,
                        ~~~~~^~~
                             max
```

This results in compilation errors, when these files get included.

Of course, user can include these headers on his side, but adding it here seems more sound.

--- a/include/nil/crypto3/zk/snark/arithmetization/plonk/constraint.hpp
+++ b/include/nil/crypto3/zk/snark/arithmetization/plonk/constraint.hpp
@@ -37,6 +37,8 @@
 #include <nil/crypto3/zk/math/expression_visitors.hpp>
 #include <nil/crypto3/zk/math/expression_evaluator.hpp>
 
+#include <map>
+
 namespace nil {
     namespace crypto3 {
         namespace zk {

--- a/include/nil/crypto3/zk/snark/arithmetization/plonk/variable.hpp
+++ b/include/nil/crypto3/zk/snark/arithmetization/plonk/variable.hpp
@@ -34,6 +34,7 @@
 #include <ostream>
 #include <vector>
 #include <functional>
+#include <map>
 #include <boost/functional/hash.hpp>
 
 namespace nil {

